app/Services/PostService.php

```diff
@@ -24,7 +24,10 @@ class PostService
      */
     public function create($data, $signupId, $transactionId)
     {
-        $this->resolvePostRepository($data['event_type']);
+        // Get the event type (type is anything after _).
+        $type = explode('_', $data['event_type'])[1];
```

Comment:
Does this always happen before calling `resolvePostRepository`? Maybe it should happen inside that function